health/tests/health_check.rs

```diff
@@ -10,14 +10,42 @@ use std::time::Duration;
 
 const TEST_SERVICE: &str = "grpc.test.TestService";
 
+#[cfg(feature = "protobuf-codec")]
+mod constants {
+    use crate::proto::ServingStatus;
+    pub const SERVING: ServingStatus = ServingStatus::Serving;
+    pub const NOT_SERVING: ServingStatus = ServingStatus::NotServing;
+    pub const SERVICE_UNKNOWN: ServingStatus = ServingStatus::ServiceUnknown;
+    pub const UNKNOWN: ServingStatus = ServingStatus::Unknown;
+}
+
+#[cfg(feature = "protobufv3-codec")]
+mod constants {
+    use grpcio_health::ServingStatus;
+    pub const SERVING: ServingStatus = ServingStatus::SERVING;
+    pub const NOT_SERVING: ServingStatus = ServingStatus::NOT_SERVING;
+    pub const SERVICE_UNKNOWN: ServingStatus = ServingStatus::SERVICE_UNKNOWN;
+    pub const UNKNOWN: ServingStatus = ServingStatus::UNKNOWN;
+}
+
+#[cfg(feature = "protobuf-codec")]
+fn response_status_equals(resp: HealthCheckResponse, status: ServingStatus) -> bool {
+    resp.status == status
+}
+
+#[cfg(feature = "protobufv3-codec")]
+fn response_status_equals(resp: HealthCheckResponse, status: ServingStatus) -> bool {
+    resp.status.enum_value().unwrap() == status
+}
+
 #[track_caller]
 fn assert_status(status: ServingStatus, client: &HealthClient, name: &str) {
     let req = HealthCheckRequest {
         service: name.to_string(),
         ..Default::default()
     };
-    let resp = client.check(&req).unwrap();
-    assert_eq!(resp.status, status)
+    let resp: HealthCheckResponse = client.check(&req).unwrap();
+    assert_eq!(response_status_equals(resp, status), true);
```

Comment:
Should be `assert!(response_status_equals(resp, status)`.

Reply:
Done